File: works/2847.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.Collections;

public class Main
{
    public static void main(String[] args) throws IOException
    {
        Scanner sc = new Scanner(System.in);
        int[] arr = new int[101];

        int N = sc.nextInt();
        for(int i = 0; i < N; i++)
        {
            int num = sc.nextInt();
            arr[i] = num;
        }

        int answer = 0;
        for(int i = N-2; i >= 0; i--)
        {
            if(arr[i] >= arr[i+1])
            {
                int sub = arr[i] - arr[i+1] + 1;
                answer += sub;
                arr[i] -= sub;
            }
        }
        System.out.print(answer);
    }
}

File: works/2847.kt
import java.util.*

fun main()
{
    val sc = Scanner(System.`in`)
    var arr = Array(101) {0}
    var N = sc.nextInt()
    for(i in 0 until N)
    {
        val num = sc.nextInt()
        arr[i] = num
    }

    var answer = 0
    for(i in N-2 downTo 0)
    {
        if(arr[i] >= arr[i+1])
        {
            var sub = arr[i] - arr[i+1] + 1
            answer += sub
            arr[i] -= sub
        }
    }
    print(answer)
}